chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/ProcessPeaksByPeakWidthFilter.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Lablicate GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Alexander Stark - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.chemclipse.xxd.model.filter.peaks;
+
+import java.util.Collection;
+import java.util.Comparator;
+import org.eclipse.chemclipse.model.core.IPeak;
+import org.eclipse.chemclipse.model.core.IPeakModel;
+import org.eclipse.chemclipse.model.filter.IPeakFilter;
+import org.eclipse.chemclipse.processing.Processor;
+import org.eclipse.chemclipse.processing.core.MessageConsumer;
+import org.eclipse.chemclipse.processing.filter.CRUDListener;
+import org.eclipse.chemclipse.processing.filter.Filter;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.osgi.service.component.annotations.Component;
+
+@Component(service = { IPeakFilter.class, Filter.class, Processor.class })
+public class ProcessPeaksByPeakWidthFilter implements IPeakFilter<ProcessPeaksByPeakWidthFilterSettings> {
+
+	@Override
+	public String getName() {
+
+		return "Peak Width Filter";
+	}
+
+	@Override
+	public String getDescription() {
+
+		return "Filter peaks by peak Width";
+	}
+
+	@Override
+	public Class<ProcessPeaksByPeakWidthFilterSettings> getConfigClass() {
+
+		return ProcessPeaksByPeakWidthFilterSettings.class;
+	}
+
+	@Override
+	public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
+
+		return true;
+	}
+
+	@Override
+	public <X extends IPeak> void filterIPeaks(CRUDListener<X, IPeakModel> listener, ProcessPeaksByPeakWidthFilterSettings configuration, MessageConsumer messageConsumer, IProgressMonitor monitor) throws IllegalArgumentException {
+
+		Collection<X> read = listener.read();
+		if(configuration == null) {
+			configuration = createConfiguration(read);
+		}
+		SubMonitor subMonitor = SubMonitor.convert(monitor, read.size());
+
+		for(X peak : read) {
+			applySelectedOptions(configuration, listener, peak);
+			subMonitor.worked(1);
+		}
+	}
+
+	private static <X extends IPeak> void applySelectedOptions(ProcessPeaksByPeakWidthFilterSettings settings, CRUDListener<X, IPeakModel> listener, X peak) {
+
+		switch (settings.getFilterSelectionCriterion()){
+		case WIDTH_SMALLER_THAN_LIMIT:
+			if(widthComparator.compare(peak.getPeakModel().getWidthByInflectionPoints(), getWidthInMilliseconds(settings))<0) {
+				deleteOrDisablePeak(listener, settings, peak);
+			}
+			break;
+		case WIDTH_GREATER_THAN_LIMIT:
+			if(widthComparator.compare(peak.getPeakModel().getWidthByInflectionPoints(), getWidthInMilliseconds(settings))>0) {
```

Comment:
Why not directly call Integer.compare()?

Reply:
If I use Double/Integer.compare() I get asked - Why not using a comparator? But if you think that this fits better, I'll gladly take it over :)

Reply:
Your comparator delegates to Integer.compare, so you can as well use Integer.compare directly. Please try to keep your comments factual.

Reply:
OK, I understand, thanks. And I will.

Reply:
BTW with java8 lamda it is possible to "implement" the comparator Inferface by simply passing Integer::compareTo to a function requiring a comparator, reverse order can be archived using Collections.reversOrder(Integer::compareTo), so I see benefits in using Interfaces Comparaor instead of the case WIDTH_SMALLER_THAN_LIMIT/.../ and simply passing DIFFERENT(!) comparators to a function depending on settings!